Fast export feature (md2html) ready to merge

fmount opened the pull request:
Since we need to enable the cbox mappings in different scenarios (e.g. :e *.notes), the cbox mapping is moved in a init function, so we can handle better the init phase when *BufRead* and *BufNewFile* events occur.
The feature is also rebased on master and all the core functions are moved in autoload section.

saro commented:
as discussed I would move the keybindings to add checkbox elements to something more specific like `<leader>[` and/or `<leader>]`. May be we can use the first to have it inline and the second for below. Also I will see how this plays when the `<leader>` is set to `<Space>.

saro commented:
Additionally, what do you think about introducing a variable to enable/disable the plugin keybindings? This could be used by user that wish to just the bindings they need.

fmount commented:
1. Added bool to enable/disable the default plugin key mappings: as described in the README section users can now use their own keybindings to define how to add/remove cbox(es)
2. keybindings moved from `<leader`>n* to `<leader`>[* (that is more specific to the *.notes) 
3. README improved with a "keybindings override" example section (including latest exposed parameters)
4. Added a separed TODO file
5. CONTRIB file renamed to reflect the most common project structure patterns
